frontend/public/components/auto-complete.tsx

```diff
@@ -0,0 +1,126 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import * as classNames from 'classnames';
+import { useDocumentListener, getLabelsAsString } from '@console/shared';
+import { fuzzyCaseInsensitive } from './factory/table-filters';
+import { TextInput } from '@patternfly/react-core';
+import { KeyEventModes } from '@console/shared/src/hooks/component-visible';
+
+const MAX_SUGGESTIONS = 5;
+
+const labelParser = (nodes) => {
+  const labels = _.map(nodes, getLabelsAsString);
+  return _.uniq(_.flatten(labels));
```

Comment:
I'm unclear, isn't `getLabelsAsString` already doing this work?

Reply:
it does for a single node at a time. We do this for multiple nodes and we flatten the array and get unique labels amongst all of them.

Reply:
> it does for a single node at a time. We do this for multiple nodes and we flatten the array and get unique labels amongst all of them.

We should use a `Set`. This is exactly what it's for.

Reply:
I could do that, but I would again have to again convert it to array for using methods like filter and slice. IMO this would make working with these values easier. WDYT?

Reply:
> I could do that, but I would again have to again convert it to array for using methods like filter and slice. IMO this would make working with these values easier. WDYT?

It's really easy to call `[...labels].filter()`, and it makes the code easier and more performant here because you don't need to worry about duplicates.